models/DataObject/ObjectAwareFieldWithMarkAsDirtyInterface.php

```diff
@@ -0,0 +1,25 @@
+<?php
+declare(strict_types=1);
+
+/**
+ * Pimcore
+ *
+ * This source file is available under two different licenses:
+ * - GNU General Public License version 3 (GPLv3)
+ * - Pimcore Commercial License (PCL)
+ * Full copyright and license information is available in
+ * LICENSE.md which is distributed with this source code.
+ *
+ *  @copyright  Copyright (c) Pimcore GmbH (http://www.pimcore.org)
+ *  @license    http://www.pimcore.org/license     GPLv3 and PCL
+ */
+
+namespace Pimcore\Model\DataObject;
+
+interface ObjectAwareFieldWithMarkAsDirtyInterface
```

Comment:
We don't need a dedicated interface here, we just use `ObjectAwareFieldInterface` and change the method signature accordingly. 
The `$markAsDirty` parameter is not needed. In case it's necessary we can call `markAllLanguagesAsDirty()` directly instead. 

Additionally we need to check why `$markAsDirty` does even exists, because in theory, changing the owner of the localized field should make any difference in the dirty status. So I guess this doesn't have a valid ready at all (need to be checked). 

See also this comment: https://github.com/pimcore/pimcore/pull/14089#discussion_r1092977900